src/main/java/org/elasticsearch/index/gateway/none/NoneIndexShardGateway.java

```diff
@@ -53,7 +53,7 @@ public String toString() {
 
     @Override
     public void recover(boolean indexShouldExists, RecoveryState recoveryState) throws IndexShardGatewayRecoveryException {
-        recoveryState.getIndex().startTime(System.currentTimeMillis());
+        indexShard.prepareForStoreRecovery();
```

Comment:
I personally would like to name these `startStoreRecovery` instead of `prepare...` since it implies that we are changing the state in that method to Stage.INDEX?